dataflow/src/main/java/org/checkerframework/dataflow/cfg/builder/CFGTranslationPhaseOne.java

```diff
@@ -1353,13 +1382,79 @@ protected List<Node> convertCallArguments(
     } else {
       for (int i = 0; i < numActuals; i++) {
         Node actualVal = scan(actualExprs.get(i), null);
+        if (i == assertMethodTuple.booleanParam) {
+          treatMethodAsAssert((MethodInvocationTree) tree, assertMethodTuple, actualVal);
+        }
         convertedNodes.add(methodInvocationConvert(actualVal, formals.get(i)));
       }
     }
 
     return convertedNodes;
   }
 
+  /**
+   * Returns the AssertMethodTuple for {@code method}. If {@code method} is not an assert method,
+   * then {@link AssertMethodTuple#NONE} is returned.
+   *
+   * @param method a method element that might be an assert method
+   * @return the AssertMethodTuple for {@code method}
+   */
+  protected AssertMethodTuple getAssertMethodTuple(ExecutableElement method) {
```

Comment:
>FYI we implemented something similar to this for NullAway in https://github.com/uber/NullAway/pull/608, via customized CFG translation. Maybe not for this PR, but it'd be cool if a client of dataflow could add to the set of methods to be treated in this manner, even if the methods don't have the appropriate annotation. This could be done, e.g., by having an overridable method in CFGTranslationPhaseOne that checks whether a method should be treated as an assertion method.

@msridhar subclasses can override this method to add addtional methods that should be treated as asserts.